packages/test/test-end-to-end-tests/src/test/summaries.spec.ts

```diff
@@ -307,6 +323,82 @@ describeNoCompat("Summaries", (getTestObjectProvider) => {
 		);
 	});
 
+	it("full initialization of data object should not happen by default", async () => {
+		const TestDataObjectType1 = "@fluid-example/test-dataStore1";
+		const dataStoreFactory1 = new DataObjectFactory(
+			TestDataObjectType1,
+			TestDataObject1,
+			[],
+			[],
+		);
+		const registryStoreEntries = new Map<string, Promise<IFluidDataStoreFactory>>([
+			[dataStoreFactory1.type, Promise.resolve(dataStoreFactory1)],
+		]);
+		const containerRuntimeFactoryWithDefaultDataStore =
+			getContainerRuntimeApi(pkgVersion).ContainerRuntimeFactoryWithDefaultDataStore;
+		const runtimeFactory = new containerRuntimeFactoryWithDefaultDataStore(
+			dataStoreFactory1,
+			registryStoreEntries,
+			undefined,
+			[],
+		);
+
+		// Create a container for the first client.
+		const container1 = await provider.createContainer(runtimeFactory);
+		assert.doesNotThrow(
+			async () => requestFluidObject<TestDataObject1>(container1, "/"),
+			"Initial creation of container and data store should succeed.",
+		);
+
+		// Create a summarizer for the container and do a summary shouldn't throw.
+		const createSummarizerResult = await createSummarizerFromFactory(
+			provider,
+			container1,
+			dataStoreFactory1,
+			undefined,
+			containerRuntimeFactoryWithDefaultDataStore,
+			registryStoreEntries,
+		);
+		const summarizer = createSummarizerResult.summarizer;
+		assert.doesNotThrow(async () => {
+			await summarizeNow(summarizer, "test");
+		});
+
+		// In summarizer, load the data store should fail.
+		const requestHeader = {
+			[LoaderHeader.clientDetails]: {
+				capabilities: { interactive: false },
+				type: "summarizer",
+			},
+			[DriverHeader.summarizingClient]: true,
+		};
+		const summarizerContainer = await provider.loadContainer(
```

Comment:
You don't need to create a summarizerContainer separately. The return from  `createSummarizerFromFactory` (createSummarizerResult) already has a summarizer container.

Reply:
How to get the summarizer container to load data object from createSummarizerResult. This loadContainer doesn't create the summarizer container again, it loads the one created before.

Reply:
I am not sure what you mean. You can do `requestFluidObject<TestDataObject1>(createSummarizerResult.container, "/")`

Reply:
`loadContainer` actually creates a new instance of the container. It says `load` in the name because its loading an existing document. Basically, its mimicing a new client that joins a Fluid document. The new clients has to create everything including container even though its "loading" a document.